### PR TITLE
CHK-1938: log enrichment

### PR DESCRIPTION
**Pull request by infantesimone**
#### Motivation and Context

The goal of the PR is to enrich the application logs in order to make them more comprehensive, therefore better describing the business logic of the transaction-service APIs.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/controllers/v2/TransactionsController.java [hunk 1]
```diff
@@ -70,11 +70,12 @@ public Mono<ResponseEntity<NewTransactionResponseDto>> newTransaction(
         return newTransactionRequest
                 .flatMap(ntr -> {
                     log.info(
-                            "newTransaction rptIDs {} ",
+                            "Create new Transaction for rptId: {}. ClientId: {} ",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v2/TransactionsController.java:73]: what about using [] braces around log values also for this and above logs too?

**Review by pietro-tota — APPROVED** [review]
Looks good, just a consideration about log formatting to make all logs being formatted the same way :)

File: src/main/java/it/pagopa/transactions/controllers/v2/TransactionsController.java (hunk 1 above)
Comment (infantesimone) [reply src/main/java/it/pagopa/transactions/controllers/v2/TransactionsController.java:73]: great!

Comment (infantesimone) [reply src/main/java/it/pagopa/transactions/controllers/v2/TransactionsController.java]: done

File: src/main/java/it/pagopa/transactions/client/NodeForPspClient.java [hunk 2]
```diff
@@ -77,24 +81,25 @@ public Mono<ActivatePaymentNoticeV2Response> activatePaymentNoticeV2(
                 .bodyToMono(ActivatePaymentNoticeV2Response.class)
                 .doOnSuccess(
                         activateResponse -> log.info(
-                                "Payment activated with paymentToken {} ",
+                                "ActivatePaymentNoticeV2 completed for noticeNumber {} ; paymentToken {} ",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/client/NodeForPspClient.java:84]:
```suggestion
                                "ActivatePaymentNoticeV2 completed for noticeNumber [{}], paymentToken [{}]",
```

File: src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java [hunk 3]
```diff
@@ -78,11 +78,12 @@ public Mono<ResponseEntity<NewTransactionResponseDto>> newTransaction(
         return newTransactionRequest
                 .flatMap(ntr -> {
                     log.info(
-                            "newTransaction rptIDs {} ",
+                            "Create new Transaction for rptId: {}. ClientId: {} ",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java:81]:
```suggestion
                            "Create new Transaction for rptId: [{}]. ClientId: [{}]",
```

File: src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java [hunk 4]
```diff
@@ -105,7 +106,7 @@ public Mono<ResponseEntity<TransactionInfoDto>> getTransactionInfo(
                                                                        ServerWebExchange exchange
     ) {
         return transactionsService.getTransactionInfo(transactionId)
-                .doOnNext(t -> log.info("getTransactionInfo for transactionId: {} ", transactionId))
+                .doOnNext(t -> log.info("GetTransactionInfo for transactionId completed: {} ", transactionId))
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java:109]:
```suggestion
                .doOnNext(t -> log.info("GetTransactionInfo for transactionId completed: [{}]", transactionId))
```

File: src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java [hunk 5]
```diff
@@ -126,7 +127,7 @@ public Mono<ResponseEntity<RequestAuthorizationResponseDto>> requestTransactionA
                                                                                                  ServerWebExchange exchange
     ) {
         return requestAuthorizationRequestDto
-                .doOnNext(t -> log.info("requestTransactionAuthorization for transactionId: {} ", transactionId))
+                .doOnNext(t -> log.info("RequestTransactionAuthorization for transactionId: {} ", transactionId))
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java:130]:
```suggestion
                .doOnNext(t -> log.info("RequestTransactionAuthorization for transactionId: [{}]", transactionId))
```

File: src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java [hunk 6]
```diff
@@ -154,7 +155,7 @@ public Mono<ResponseEntity<TransactionInfoDto>> updateTransactionAuthorization(
                 transactionIdDecoded -> updateAuthorizationRequestDto
                         .doOnNext(
                                 t -> log.info(
-                                        "updateTransactionAuthorization for transactionId: {}, decoded transaction id: {} ",
+                                        "UpdateTransactionAuthorization for transactionId: {}, decoded transaction id: {} ",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java:158]:
```suggestion
                                        "UpdateTransactionAuthorization for transactionId: [{}], decoded transaction id: [{}]",
```

File: src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java [hunk 7]
```diff
@@ -222,7 +223,7 @@ public Mono<ResponseEntity<AddUserReceiptResponseDto>> addUserReceipt(
                                                                           ServerWebExchange exchange
     ) {
         return addUserReceiptRequestDto
-                .doOnNext(t -> log.info("addUserReceipt for transactionId: {} ", transactionId))
+                .doOnNext(t -> log.info("AddUserReceipt for transactionId: {} ", transactionId))
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/controllers/v1/TransactionsController.java:226]:
```suggestion
                .doOnNext(t -> log.info("AddUserReceipt for transactionId: [{}]", transactionId))
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 8]
```diff
@@ -712,7 +712,7 @@ public Mono<TransactionInfoDto> updateTransactionAuthorization(
     ) {
 
         TransactionId transactionId = new TransactionId(decodedTransactionId);
-        log.info("decoded transaction id: {}", transactionId.value());
+        log.info("UpdateTransactionAuthorization decoded transaction id: {}", transactionId.value());
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:715]:
```suggestion
        log.info("UpdateTransactionAuthorization decoded transaction id: [{}]", transactionId.value());
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 9]
```diff
@@ -800,7 +800,7 @@ private Mono<TransactionInfoDto> updateTransactionAuthorizationStatusV1(
                         return baseTransaction.map(
                                 trx -> {
                                     log.info(
-                                            "Transaction authorization outcome already received. Transaction status: {}",
+                                            "UpdateTransactionAuthorization Transaction authorization outcome already received. Transaction status: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:803]:
```suggestion
                                            "UpdateTransactionAuthorization Transaction authorization outcome already received. Transaction status: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 10]
```diff
@@ -817,14 +817,14 @@ private Mono<TransactionInfoDto> updateTransactionAuthorizationStatusV1(
                                                         .handle(transactionUpdateAuthorizationCommand)
                                                         .doOnNext(
                                                                 authorizationStatusUpdatedEvent -> log.info(
-                                                                        "Requested authorization update for rptId: {}",
+                                                                        "UpdateTransactionAuthorization Requested authorization update for rptId: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:820]:
```suggestion
                                                                        "UpdateTransactionAuthorization Requested authorization update for rptId: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 11]
```diff
@@ -817,14 +817,14 @@ private Mono<TransactionInfoDto> updateTransactionAuthorizationStatusV1(
...
                                                                         transaction.getPaymentNotices().get(0).rptId()
                                                                 )
                                                         )
                                                         .doOnError(
                                                                 AlreadyProcessedException.class,
                                                                 exception -> log.error(
-                                                                        "Error: requesting authorization update for transaction in state {}",
+                                                                        "UpdateTransactionAuthorization Error: requesting authorization update for transaction in state {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:827]:
```suggestion
                                                                        "UpdateTransactionAuthorization Error: requesting authorization update for transaction in state [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 12]
```diff
@@ -880,7 +880,7 @@ private Mono<TransactionInfoDto> updateTransactionAuthorizationStatusV2(
                         return baseTransaction.map(
                                 trx -> {
                                     log.info(
-                                            "Transaction authorization outcome already received. Transaction status: {}",
+                                            "UpdateTransactionAuthorization outcome already received. Transaction status: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:883]:
```suggestion
                                            "UpdateTransactionAuthorization outcome already received. Transaction status: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 13]
```diff
@@ -898,14 +898,14 @@ private Mono<TransactionInfoDto> updateTransactionAuthorizationStatusV2(
                                                         .handle(transactionUpdateAuthorizationCommand)
                                                         .doOnNext(
                                                                 authorizationStatusUpdatedEvent -> log.info(
-                                                                        "Requested authorization update for rptId: {}",
+                                                                        "UpdateTransactionAuthorization requested for rptId: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:901]:
```suggestion
                                                                        "UpdateTransactionAuthorization requested for rptId: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 14]
```diff
@@ -898,14 +898,14 @@ private Mono<TransactionInfoDto> updateTransactionAuthorizationStatusV2(
...
                                                                         transaction.getPaymentNotices().get(0).rptId()
                                                                 )
                                                         )
                                                         .doOnError(
                                                                 AlreadyProcessedException.class,
                                                                 exception -> log.error(
-                                                                        "Error: requesting authorization update for transaction in state {}",
+                                                                        "UpdateTransactionAuthorization Error: requesting authorization update for transaction in state {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:908]:
```suggestion
                                                                        "UpdateTransactionAuthorization Error: requesting authorization update for transaction in state [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 15]
```diff
@@ -1141,7 +1141,7 @@ public Mono<TransactionInfoDto> addUserReceipt(
                                             .handle(el._2)
                                             .doOnNext(
                                                     transactionUserReceiptRequestedEvent -> log.info(
-                                                            "{} for transactionId: {}",
+                                                            "AddUserReceipt {} for transactionId: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:1144]:
```suggestion
                                                            "AddUserReceipt [{}] for transactionId: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 16]
```diff
@@ -1150,7 +1150,7 @@ public Mono<TransactionInfoDto> addUserReceipt(
                                                     .handle((TransactionUserReceiptRequestedEvent) event))
                                             .doOnNext(
                                                     transaction -> log.info(
-                                                            "Transaction status updated {} for transactionId: {}",
+                                                            "AddUserReceipt transaction status updated {} for transactionId: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:1153]:
```suggestion
                                                            "AddUserReceipt transaction status updated [{}] for transactionId: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 17]
```diff
@@ -1161,7 +1161,7 @@ public Mono<TransactionInfoDto> addUserReceipt(
                                     .handle(el._2)
                                     .doOnNext(
                                             transactionUserReceiptRequestedEvent -> log.info(
-                                                    "{} for transactionId: {}",
+                                                    "AddUserReceipt {} for transactionId: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:1164]:
```suggestion
                                                    "AddUserReceipt [{}] for transactionId: [{}]",
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 18]
```diff
@@ -1170,7 +1170,7 @@ public Mono<TransactionInfoDto> addUserReceipt(
                                             .handle((it.pagopa.ecommerce.commons.documents.v2.TransactionUserReceiptRequestedEvent) event))
                                     .doOnNext(
                                             transaction -> log.info(
-                                                    "Transaction status updated {} for transactionId: {}",
+                                                    "AddUserReceipt transaction status updated {} for transactionId: {}",
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:1173]:
```suggestion
                                                    "AddUserReceipt transaction status updated [{}] for transactionId: [{}]",
```

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**